c2rust-analyze/src/main.rs

```diff
@@ -474,6 +474,24 @@ fn run(tcx: TyCtxt) {
                 || info.contains(PointerInfo::NOT_TEMPORARY_REF))
     }
 
+    // track all types mentioned in extern blocks, we
+    // don't want to rewrite those
+    let mut foreign_mentioned_tys = HashSet::new();
+    for item in tcx.hir_crate_items(()).foreign_items() {
+        let ldid = item.def_id.to_def_id();
+        match tcx.def_kind(ldid) {
+            DefKind::Fn | DefKind::AssocFn => {
```

Comment:
Why are we only looking at functions?  What about statics, for example, or whatever else there may be?

Reply:
I thought about statics but I wasnt sure, could you give an example of a declaration item in the extern block youre thinking of and I can add support for it?

Reply:
Here's an example involving an extern static with struct type:

```C
struct S {
    int x;
    int y;
};

extern S s;

int f() { return s.x; }
```

```Rust
extern "C" {
    static mut s: S;
}
#[derive(Copy, Clone)]
#[repr(C)]
pub struct S {
    pub x: libc::c_int,
    pub y: libc::c_int,
}
#[no_mangle]
pub unsafe extern "C" fn f() -> libc::c_int {
    return s.x;
}
```

Reply:
Extern types don't have any rewritable parts, so we don't actually need to do any work in that case.

Reply:
Yeah, that's true.  I think they'd also be mentioned in `extern` `fn`s` and `static`s anyways.

Reply:
Added in 9f35bbfe